Johnson_Algorithm/Johnson's_Algorithm_Readme.md

```diff
@@ -0,0 +1,54 @@
+# Johnson's Algorithm
+
+Johnson's algorithm is a method to find the shortest distance between every pair of vertices in a weighted directed graph. The edges weights may be negative, but no negative cycles should be present in the graph.
+Johnson's algorithm utilizes both Dijkstra's algorithm and Bellman-Ford's algorithm as subroutines. The algorithm can be applied on sparse graphs, and the time complexity depends on the number of edges.
+Johnson's algorithm uses the technique of reweighting, which works as follows:
+
+If all edge weights w in a graph G = (V, E) are nonnegative, we can find shortest paths between all pairs of vertices by running Dijkstra's algorithm once from each vertex.
+
+If G has negative-weight edges but no negative-weight cycles, we simply compute a new set of nonnegative edge weights that allows us to use the same method. The new set of edge weights must satisfy two important properties.
+1. For all pairs of vertices u, v V, a path p is the shortest path from u to v using weight function w if and only if p is also the shortest path from u to v using weight function.
+2. For all edges (u, v), the new weight is nonnegative.
+
+## Algorithm
+
```

Comment:
Where is the algorithm?

Reply:
The input output and pseudo code. The algorithm is split into those 3 parts